cds/modules/maintenance/cli.py

```diff
@@ -38,13 +37,57 @@
     create_subformat,
 )
 from cds.modules.records.api import CDSVideosFilesIterator
+from cds.modules.deposit.api import Video
 from cds.modules.records.resolver import record_resolver
+from cds.modules.records.minters import doi_minter
+from cds.modules.deposit.tasks import datacite_register
+from invenio_db import db
+from cds.modules.records.permissions import is_public
+from invenio_indexer.api import RecordIndexer
+
 
 
 def abort_if_false(ctx, param, value):
     if not value:
         ctx.abort()
 
+@click.command()
+@click.option("--recid", "recid", help="ID of the video record", default=None)
+@with_appcontext
+def create_doi(recid):
+    if not recid:
+        raise ClickException('Missing option "--recid')
+    
+    try:
+        # Get the video object with recid
+        _, record = record_resolver.resolve(recid)
+        depid = record.depid
+        video_deposit = Video.get_record(depid.object_uuid)
+    except Exception as exc:
+        raise ClickException("Failed to fetch the video record")
+
+    if is_public(video_deposit, "read") and video_deposit.is_published():
+        try:
+            # sync deposit files <--> record files
+            edit_record = video_deposit.edit()
+            doi_minter(record_uuid=edit_record.id, data=edit_record)
+            edit_record.publish().commit()
+
+            # save changes
+            db.session.commit()
+
+            # Register the doi to datacite 
+            datacite_register(recid, str(record.id))
+        except Exception as exc:
+            db.session.rollback()
+        # index the record again
+        _, record_video = edit_record.fetch_published()
```

Comment:
these methods should be moved inside the `Exception` block because otherwise we issue a redundant reindex of the record after the `db.session.rollback()` above.